apache/superset
# feat: [SIP-114] Google Sheets Export (part 1)

### SUMMARY

**This PR implements https://github.com/apache/superset/issues/26390, providing one-click Google Sheets Export.**

I've marked this as a WIP but it's a fully working implementation, missing is:
- [x] Need to look into the documentation system.

I've split the SIP implementation into two PRs:
- (This PR) Export route, lib code, SqlLab export implementation.
- Adding Google Sheets to Alerts and Reports.




### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF


**Before:**
![Screenshot 2024-01-02 at 5 08 30 pm](https://github.com/apache/superset/assets/971886/ad16668e-4176-4a73-b56b-d818752e85eb)

**After**
![Screenshot 2024-01-02 at 5 08 43 pm](https://github.com/apache/superset/assets/971886/62910e16-c3ad-4f5d-a85f-7a0bd2505689)

**Video Demo**

https://github.com/apache/superset/assets/971886/78504a9f-727e-4a9e-a278-bb58a547f025


### TESTING INSTRUCTIONS

See the added docs:
1. Follow setup instructions in the docs.
2. Run a query in SqlLab.
3. Click the "Google Sheets" button under the new "Export" dropdown.

### ADDITIONAL INFORMATION


- [x] Has associated issue: https://github.com/apache/superset/issues/26390
   - `GOOGLE_SHEETS_EXPORT`
